### ISSUE TITLE
Missing resources for lib2to3 and main package

### ISSUE THREAD

**Issue by golyalpha** [Nuitka/Nuitka#496]
Nuitka installation evironment: see #492 

Nuitka build command:
```
poetry run python -m nuitka --standalone --python-flag=no_site --show-progress --show-scons blender_renderfarm
```
Crash traceback:
```
Traceback (most recent call last):
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\__main__.py", line 20, in <module>
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\fire\core.py", line 138, in Fire
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\fire\core.py", line 466, in _Fire
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\fire\core.py", line 675, in _CallAndUpdateTrace
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\__main__.py", line 9, in render
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\blender_renderfarm\__init__.py", line 4, in <module blender_renderfarm>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\blender_renderfarm\encoder.py", line 1, in <module blender_renderfarm.encoder>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\ffmpeg\__init__.py", line 2, in <module ffmpeg>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\ffmpeg\nodes.py", line 3, in <module ffmpeg.nodes>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\past\__init__.py", line 88, in <module past>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\past\translation\__init__.py", line 42, in <module past.translation>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\lib2to3\refactor.py", line 25, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\lib2to3\fixer_util.py", line 7, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\lib2to3\pygram.py", line 32, in <module>
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\lib2to3\pgen2\driver.py", line 158, in load_packaged_grammar
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\pkgutil.py", line 637, in get_data
  File "<frozen importlib._bootstrap_external>", line 916, in get_data
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\git\\blender-renderfarm\\blender_renderfarm.dist\\lib2to3\\Grammar3.7.4.final.0.pickle'
```

Note: It is possible that the above is a bug connected with using Nuitka inside venvs, as I was unable to find these resources in my venv, and had to look for them in the system interpreter files instead. The source runs fine inside the venv.

Alongside the issue of some dependencies missing resources, the main package is missing them as well:
```
> .\blender_renderfarm.exe benchmark
Preparing for benchmark.
Traceback (most recent call last):
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\__main__.py", line 20, in <module>
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\fire\core.py", line 138, in Fire
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\fire\core.py", line 466, in _Fire
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\fire\core.py", line 675, in _CallAndUpdateTrace
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\__main__.py", line 17, in benchmark
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\blender_renderfarm\__init__.py", line 16, in construct_farmspec
  File "D:\git\blender-renderfarm\blender_renderfarm.dist\pkgutil.py", line 637, in get_data
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\git\\blender-renderfarm\\blender_renderfarm.dist\\blender_renderfarm\\resources\\bench.blend'
```

**Comment by kayhayen** [comment]
Actually `pkgutils.get_data` ought to be our best friend, because we will often be capable of analysing it at compile time.

That sender, renderfarm usage is not generally interesting, and you have to copy your own data files for now, although as I mentioned above, a plugin of Nuitka ought to inform you about those or statically analyse them.

As for lib2to3, is it really using that, or just happening to import it? I need to check it out, but I assume a simple addition to the data files plugin will do. Unfortunately I don't have a reproducer.

Yours,
Kay

**Comment by golyalpha** [comment]
The lib2to3 error seems to be a thing with Python Fire (the library it's using to construct the CLI), the package itself isn't using it directly.

I honestly did not understand what exactly you meant by this line:
```
That sender, renderfarm usage is not generally interesting
```
but if I understand that whole part correctly, resource files from the package have to be copied manually...?
That is something I can live with as that's easy to automate, but that lib2to3 issue (and actually I had an issue with Tk and Tcl missing it's files too) ought to be investigated more fully. I'll see what I can do to reproduce the issues.

**Comment by golyalpha** [comment]
Ok, Nuitka successfuly warns if Tk is imported in the main_module file, but it does not warn at all, if Tk is imported inside a call in a separate file, even if all main_module does is import that call from that other file and calls it.


tk_test.py
```py
from tk_test2 import main

main()
```

tk_test2.py
```py
def main():
    from tkinter import Tk

    Tk().withdraw()
```

Crash output:
```
> .\tk_test.dist\tk_test.exe
Traceback (most recent call last):
  File "D:\git\blender-renderfarm\tk_test.dist\tk_test.py", line 3, in <module>
  File "D:\git\blender-renderfarm\tk_test.dist\tk_test2.py", line 4, in main
  File "D:\git\blender-renderfarm\tk_test.dist\tkinter\__init__.py", line 2023, in __init__
_tkinter.TclError: Can't find a usable init.tcl in the following directories:
    D:/git/blender-renderfarm/lib/tcl8.6 D:/git/blender-renderfarm/lib/tcl8.6 D:/git/lib/tcl8.6 D:/git/blender-renderfarm/library D:/git/library D:/git/tcl8.6.9/library D:/tcl8.6.9/library



This probably means that Tcl wasn't installed properly.
```

**Comment by golyalpha** [comment]
And no warning from Nuitka in relation with lib2to3

l23_test.py
```py
import lib2to3.pgen2.driver

lib2to3.pgen2.driver.load_packaged_grammar("lib2to3", "Grammar")
```

Crash output:
```
> .\l23_test.dist\l23_test.exe
Traceback (most recent call last):
  File "D:\git\blender-renderfarm\l23_test.dist\l23_test.py", line 3, in <module>
  File "D:\git\blender-renderfarm\l23_test.dist\lib2to3\pgen2\driver.py", line 158, in load_packaged_grammar
  File "D:\git\blender-renderfarm\l23_test.dist\pkgutil.py", line 637, in get_data
  File "<frozen importlib._bootstrap_external>", line 916, in get_data
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\git\\blender-renderfarm\\l23_test.dist\\lib2to3\\Grammar3.7.4.final.0.pickle'
```

**Comment by kayhayen** [comment]
There is a fix for this on the factory branch. 

https://nuitka.net/doc/factory.html

**Comment by golyalpha** [comment]
By the way, Poetry installs the project into the venv as an editable dependency. Shouldn't that help Nuitka bundle resource files into the resulting dist?

**Comment by kayhayen** [comment]
Part of the latest pre-release rc6.

**Comment by golyalpha** [comment]
@kayhayen lib2to3 or package resources?

**Comment by kayhayen** [comment]
Released as part of 0.6.6
